### PR TITLE
fix: shift due dates banner overlap

### PR DESCRIPTION
**Pull request by forgotvas**
[iOS] The loading indicator overlaps “Shift due dates” button #419

How it works now: 

https://github.com/user-attachments/assets/26a8c2eb-599f-48c4-8620-aa5cc0e318bf

Also that PR contains fix for profile icon issue on profile tab on iOS 15 it disappears when you go to settings and then go back to profile

### CODE REVIEW

**Review by rnr — APPROVED** [review]

File: OpenEdX/View/MainScreenView.swift [hunk 1]
```diff
@@ -188,7 +188,9 @@ struct MainScreenView: View {
         }
         .accentColor(Theme.Colors.accentXColor)
         .introspect(.viewController, on: .iOS(.v15)) { controller in
-            controller.navigationController?.setNavigationBarHidden(true, animated: false)
+            if viewModel.selection != .profile {
+                controller.navigationController?.setNavigationBarHidden(true, animated: false)
+            }
```
Comment (saeedbashir) [inline OpenEdX/View/MainScreenView.swift:193]: @forgotvas  Is this change related to the mentioned issue?

Comment (forgotvas) [reply OpenEdX/View/MainScreenView.swift:193]: that change related to profile icon issue on profile tab on iOS 15 it disappears when you go to settings and then go back to profile. So just fixed in that PR too. Added to description.

Comment (saeedbashir) [reply OpenEdX/View/MainScreenView.swift:193]: I guess the issue is also with the parsing, app is expecting, `"account_privacy": "private"` but server is returning `"account_privacy": "PRIVATE"`. 

Can you look into it under the same PR or should create a separate ticket for it. 

Credentials: `rtester97/Fedx1234`

Comment (forgotvas) [reply OpenEdX/View/MainScreenView.swift:193]: added in https://github.com/edx/edx-mobile-marketplace-ios/pull/59

**Review by saeedbashir — APPROVED** [review]
